write/batcher_test.go

```diff
@@ -304,26 +324,26 @@ func TestBatcher_write(t *testing.T) {
 
 			if cancel != nil {
 				cancel()
-			} else {
-				tt.args.lines <- []byte(tt.args.line)
-				// if the max flush interval is not zero,  we are testing to see
-				// if the data is flushed via the timer rather than forced by
-				// closing the channel.
-				if tt.fields.MaxFlushInterval != 0 {
-					time.Sleep(tt.fields.MaxFlushInterval * 100)
-				}
-				close(tt.args.lines)
+				time.Sleep(500 * time.Millisecond)
+			}
+
+			tt.args.lines <- []byte(tt.args.line)
+			// if the max flush interval is not zero, we are testing to see
+			// if the data is flushed via the timer rather than forced by
+			// closing the channel.
+			if tt.fields.MaxFlushInterval != 0 {
+				time.Sleep(tt.fields.MaxFlushInterval * 100)
 			}
+			close(tt.args.lines)
 
 			err := <-tt.args.errC
 			if (err != nil) != tt.wantErr {
 				t.Errorf("ScanLines.read() error = %v, wantErr %v", err, tt.wantErr)
 				return
 			}
 
-			if !cmp.Equal(got, tt.want) {
-				t.Errorf("%q. Batcher.write() = -got/+want %s", tt.name, cmp.Diff(got, tt.want))
-			}
+			require.Equal(t, tt.wantNoCall, !writeCalled)
```

Comment:
This logic would be a bit prettier if this had been called `wantCall`

Reply:
True! I like `wantNoCall` because the zero value is the most common case, so we can leave it unspecified in most of the test-case structs & reduce noise.